src/modules/ekf2/EKF2.cpp

```diff
@@ -1699,6 +1687,45 @@ void EKF2::PublishOdometry(const hrt_abstime &timestamp, const imuSample &imu_sa
 	_odometry_pub.publish(odom);
 }
 
+void EKF2::PublishFullState(const hrt_abstime &timestamp, const imuSample &imu_sample)
+{
+	vehicle_full_state_s fs{};
+	fs.timestamp_sample = imu_sample.time_us;
+
+	// Position (local NED)
+	const Vector3f position{_ekf.getPosition()};
+	fs.position[0] = position(0);
+	fs.position[1] = position(1);
+	fs.position[2] = position(2);
+
+	// Velocity (local NED)
+	const Vector3f velocity{_ekf.getVelocity()};
+	fs.velocity[0] = velocity(0);
+	fs.velocity[1] = velocity(1);
+	fs.velocity[2] = velocity(2);
+
+	// Linear acceleration (local NED)
+	// EKF provides velocity derivative in local frame
+	const Vector3f vel_deriv{_ekf.getVelocityDerivative()};
+	fs.acceleration[0] = vel_deriv(0);
+	fs.acceleration[1] = vel_deriv(1);
+	fs.acceleration[2] = vel_deriv(2);
+
+	// Orientation quaternion (FRD -> NED), same convention as vehicle_odometry.q
+	_ekf.getQuaternion().copyTo(fs.q); // (w,x,y,z)
+
+	// Angular velocity (body FRD), same as vehicle_odometry angular_velocity
+	const Vector3f rates{imu_sample.delta_ang / imu_sample.delta_ang_dt};
```

Comment:
You'll get aliasing if you do this. That's the reason why we have `getAngularVelocityAndResetAccumulator` in `PublishOdometry` (of course, you can't just call the same function here as it resets the accumulator). And a similar thing needs to be done on the acceleration data.